lib/GoalCohort.js

```diff
@@ -76,10 +77,19 @@ const GoalCohort = new Schema(
       required: "{PATH} is required!",
       index: true
     },
+    name: { type: String },
+    nameCanonical: {
```

Comment:
This index is going to make it so that there can only be one document with each `nameCanonical`. That's wrong because we want to have one cohort with a given name for each *goal*.

To frame the problem, create a new test that's like 

```
it('allows a cohort name to be used exactly once per goal", 
```

...and then in that test make sure you try to use. the same cohort name for two different goals

In terms of implementation, you can't. define a compound. index inside of the schema definition; you need to do it afterwards. like

```
UserCohort.index({ goal: 1, nameCanonical: 1 }, { unique: true, spare: true });
```